pkg/collector/corechecks/cluster/kubernetesapiserver/unbundled_events.go

```diff
@@ -64,63 +64,78 @@ func (c *unbundledTransformer) Transform(events []*v1.Event) ([]event.Event, []e
 		involvedObject := ev.InvolvedObject
 		hostInfo := getEventHostInfo(c.clusterName, ev)
 		readableKey := buildReadableKey(involvedObject)
-		tagsAccumulator := tagset.NewHashlessTagsAccumulator()
 
-		tagsAccumulator.Append(getInvolvedObjectTags(involvedObject, c.taggerInstance)...)
-		tagsAccumulator.Append(
-			fmt.Sprintf("source_component:%s", ev.Source.Component),
-			fmt.Sprintf("event_reason:%s", ev.Reason))
-
-		if hostInfo.providerID != "" {
-			tagsAccumulator.Append(fmt.Sprintf("host_provider_id:%s", hostInfo.providerID))
-		}
-		c.getTagsFromTagger(involvedObject, tagsAccumulator)
-		tagsAccumulator.SortUniq()
+		tags := c.buildEventTags(ev, involvedObject, hostInfo)
 
 		emittedEvents.Inc(involvedObject.Kind, ev.Type)
-
-		datadogEvs = append(datadogEvs, event.Event{
+		event := event.Event{
 			Title:          fmt.Sprintf("%s: %s", readableKey, ev.Reason),
 			Priority:       event.PriorityNormal,
 			Host:           hostInfo.hostname,
 			SourceTypeName: "kubernetes",
 			EventType:      CheckName,
 			Ts:             int64(ev.LastTimestamp.Unix()),
-			Tags:           tagsAccumulator.Get(),
+			Tags:           tags,
 			AggregationKey: fmt.Sprintf("kubernetes_apiserver:%s", involvedObject.UID),
 			AlertType:      getDDAlertType(ev.Type),
 			Text:           ev.Message,
-		})
+		}
+		datadogEvs = append(datadogEvs, event)
 	}
 
 	return datadogEvs, errors
 }
 
+// buildEventTags aggregate all tags for an event from multiple sources.
+func (c *unbundledTransformer) buildEventTags(ev *v1.Event, involvedObject v1.ObjectReference, hostInfo eventHostInfo) []string {
+	tagsAccumulator := tagset.NewHashlessTagsAccumulator()
+
+	// Hardcoded tags
+	tagsAccumulator.Append(
+		fmt.Sprintf("source_component:%s", ev.Source.Component),
```

Comment:
I think it could be empty in some cases. We can also use `ReportingController`.

Reply:
That's the way the code was before, but I guess we could be smarter/add more tags.